### PR TITLE
[Merged by Bors] - Fix crash in headless mode

### LINKED ISSUE #4440: headless_defaults example panics again
## Bevy version

Seemingly introduced in c5963b4fd5e37040b927fa19ea55be7653e1ce15

## Operating system & version

MacOS, Linux (Lavapipe)
Rust 1.59.0, 1.60.0

## What you did

```cargo run --example headless_defaults```

## What actually happened

```
thread 'Compute Task Pool (0)' panicked at 'Resource requested by bevy_pbr::light::assign_lights_to_clusters does not exist: bevy_render::renderer::render_device::RenderDevice', /Users/robparrett/src/bevy/crates/bevy_ecs/src/system/system_param.rs:319:17
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: <bevy_ecs::system::system_param::ResState<T> as bevy_ecs::system::system_param::SystemParamFetch>::get_param::{{closure}}
             at ./crates/bevy_ecs/src/system/system_param.rs:319:17
   3: core::option::Option<T>::unwrap_or_else
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/option.rs:779:21
   4: <bevy_ecs::system::system_param::ResState<T> as bevy_ecs::system::system_param::SystemParamFetch>::get_param
             at ./crates/bevy_ecs/src/system/system_param.rs:316:22
   5: <(P0,P1,P2,P3,P4,P5,P6,P7,P8) as bevy_ecs::system::system_param::SystemParamFetch>::get_param
             at ./crates/bevy_ecs/src/system/system_param.rs:1238:20
   6: <Func as bevy_ecs::system::function_system::SystemParamFunction<(),Out,(F0,F1,F2,F3,F4,F5,F6,F7,F8),()>>::run
             at ./crates/bevy_ecs/src/system/function_system.rs:513:37
   7: <bevy_ecs::system::function_system::FunctionSystem<In,Out,Param,Marker,F> as bevy_ecs::system::system::System>::run_unsafe
             at ./crates/bevy_ecs/src/system/function_system.rs:401:19
   8: bevy_ecs::schedule::executor_parallel::ParallelExecutor::prepare_systems::{{closure}}
             at ./crates/bevy_ecs/src/schedule/executor_parallel.rs:210:30
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/mod.rs:84:19
  10: async_executor::Executor::spawn::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:144:19
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/mod.rs:84:19
  12: async_task::raw::RawTask<F,T,S>::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/raw.rs:489:20
  13: async_task::runnable::Runnable::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/runnable.rs:309:18
  14: async_executor::Executor::run::{{closure}}::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:235:21
  15: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/mod.rs:84:19
  16: <futures_lite::future::Or<F1,F2> as core::future::future::Future>::poll
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-lite-1.12.0/src/future.rs:529:33
  17: async_executor::Executor::run::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:242:31
  18: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/mod.rs:84:19
  19: futures_lite::future::block_on::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-lite-1.12.0/src/future.rs:89:27
  20: std::thread::local::LocalKey<T>::try_with
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/thread/local.rs:412:16
  21: std::thread::local::LocalKey<T>::with
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/thread/local.rs:388:9
  22: futures_lite::future::block_on
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-lite-1.12.0/src/future.rs:79:5
  23: bevy_tasks::task_pool::TaskPool::new_internal::{{closure}}::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:150:25
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at 'task has failed', /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/task.rs:425:45
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: core::panicking::panic_display
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:72:5
   3: core::panicking::panic_str
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:56:5
   4: core::option::expect_failed
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/option.rs:1817:5
   5: core::option::Option<T>::expect
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/option.rs:692:21
   6: <async_task::task::Task<T> as core::future::future::Future>::poll
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/task.rs:425:43
   7: bevy_tasks::task_pool::TaskPool::scope::{{closure}}::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:205:42
   8: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/mod.rs:84:19
   9: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/future.rs:123:9
  10: async_executor::LocalExecutor::spawn::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:376:19
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/future/mod.rs:84:19
  12: async_task::raw::RawTask<F,T,S>::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/raw.rs:489:20
  13: async_task::runnable::Runnable::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-4.2.0/src/runnable.rs:309:18
  14: async_executor::Executor::try_tick
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:181:17
  15: async_executor::LocalExecutor::try_tick
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/async-executor-1.4.1/src/lib.rs:405:9
  16: bevy_tasks::task_pool::TaskPool::scope::{{closure}}
             at ./crates/bevy_tasks/src/task_pool.rs:234:21
  17: std::thread::local::LocalKey<T>::try_with
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/thread/local.rs:412:16
  18: std::thread::local::LocalKey<T>::with
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/thread/local.rs:388:9
  19: bevy_tasks::task_pool::TaskPool::scope
             at ./crates/bevy_tasks/src/task_pool.rs:180:9
  20: <bevy_ecs::schedule::executor_parallel::ParallelExecutor as bevy_ecs::schedule::executor::ParallelSystemExecutor>::run_systems
             at ./crates/bevy_ecs/src/schedule/executor_parallel.rs:122:9
  21: <bevy_ecs::schedule::stage::SystemStage as bevy_ecs::schedule::stage::Stage>::run
             at ./crates/bevy_ecs/src/schedule/stage.rs:853:17
  22: bevy_ecs::schedule::Schedule::run_once
             at ./crates/bevy_ecs/src/schedule/mod.rs:341:13
  23: <bevy_ecs::schedule::Schedule as bevy_ecs::schedule::stage::Stage>::run
             at ./crates/bevy_ecs/src/schedule/mod.rs:359:21
  24: bevy_app::app::App::update
             at ./crates/bevy_app/src/app.rs:114:9
  25: bevy_winit::winit_runner_with::{{closure}}
             at ./crates/bevy_winit/src/lib.rs:563:21
  26: <winit::platform_impl::platform::app_state::EventLoopHandler<T> as winit::platform_impl::platform::app_state::EventHandler>::handle_nonuser_event::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/app_state.rs:101:13
  27: winit::platform_impl::platform::app_state::EventLoopHandler<T>::with_callback
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/app_state.rs:79:13
  28: <winit::platform_impl::platform::app_state::EventLoopHandler<T> as winit::platform_impl::platform::app_state::EventHandler>::handle_nonuser_event
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/app_state.rs:100:9
  29: winit::platform_impl::platform::app_state::Handler::handle_nonuser_event
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/app_state.rs:206:21
  30: winit::platform_impl::platform::app_state::AppState::cleared
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/app_state.rs:387:9
  31: winit::platform_impl::platform::observer::control_flow_end_handler::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/observer.rs:180:21
  32: winit::platform_impl::platform::observer::control_flow_handler::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/observer.rs:142:57
  33: std::panicking::try::do_call
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:406:40
  34: <unknown>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/fmt/mod.rs:2293:14
  35: std::panicking::try
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:370:19
  36: std::panic::catch_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panic.rs:133:14
  37: winit::platform_impl::platform::event_loop::stop_app_on_panic
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:229:11
  38: winit::platform_impl::platform::observer::control_flow_handler
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/observer.rs:142:5
  39: winit::platform_impl::platform::observer::control_flow_end_handler
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/observer.rs:175:9
  40: <unknown>
  41: <unknown>
  42: <unknown>
  43: <unknown>
  44: <unknown>
  45: <unknown>
  46: <unknown>
  47: <unknown>
  48: <unknown>
  49: <unknown>
  50: <() as objc::message::MessageArguments>::invoke
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.2.7/src/message/mod.rs:128:17
  51: objc::message::platform::send_unverified
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.2.7/src/message/apple/mod.rs:27:9
  52: objc::message::send_message
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.2.7/src/message/mod.rs:178:5
  53: winit::platform_impl::platform::event_loop::EventLoop<T>::run_return::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:188:22
  54: objc::rc::autorelease::autoreleasepool
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.2.7/src/rc/autorelease.rs:29:5
  55: winit::platform_impl::platform::event_loop::EventLoop<T>::run_return
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:178:9
  56: winit::platform_impl::platform::event_loop::EventLoop<T>::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:157:9
  57: winit::event_loop::EventLoop<T>::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/event_loop.rs:154:9
  58: bevy_winit::run
             at ./crates/bevy_winit/src/lib.rs:180:5
  59: bevy_winit::winit_runner_with
             at ./crates/bevy_winit/src/lib.rs:604:9
  60: bevy_winit::winit_runner
             at ./crates/bevy_winit/src/lib.rs:220:5
  61: core::ops::function::Fn::call
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ops/function.rs:70:5
  62: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/boxed.rs:1868:9
  63: bevy_app::app::App::run
             at ./crates/bevy_app/src/app.rs:130:9
  64: headless_defaults::main
             at ./examples/app/headless_defaults.rs:4:5
  65: core::ops::function::FnOnce::call_once
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread 'main' panicked at 'Task thread panicked while executing.: Any { .. }', crates/bevy_tasks/src/task_pool.rs:77:21
stack backtrace:
   0: rust_begin_unwind
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
   1: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
   2: core::result::unwrap_failed
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/result.rs:1690:5
   3: core::result::Result<T,E>::expect
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/result.rs:975:23
   4: <bevy_tasks::task_pool::TaskPoolInner as core::ops::drop::Drop>::drop
             at ./crates/bevy_tasks/src/task_pool.rs:77:17
   5: core::ptr::drop_in_place<bevy_tasks::task_pool::TaskPoolInner>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
   6: alloc::sync::Arc<T>::drop_slow
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/sync.rs:1079:18
   7: <alloc::sync::Arc<T> as core::ops::drop::Drop>::drop
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/sync.rs:1640:13
   8: core::ptr::drop_in_place<alloc::sync::Arc<bevy_tasks::task_pool::TaskPoolInner>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
   9: core::ptr::drop_in_place<bevy_tasks::task_pool::TaskPool>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  10: core::ptr::drop_in_place<bevy_tasks::usages::ComputeTaskPool>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  11: core::ptr::mut_ptr::<impl *mut T>::drop_in_place
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mut_ptr.rs:1049:18
  12: bevy_ecs::component::ComponentDescriptor::drop_ptr
             at ./crates/bevy_ecs/src/component.rs:182:9
  13: bevy_ecs::storage::blob_vec::BlobVec::clear
             at ./crates/bevy_ecs/src/storage/blob_vec.rs:213:17
  14: <bevy_ecs::storage::blob_vec::BlobVec as core::ops::drop::Drop>::drop
             at ./crates/bevy_ecs/src/storage/blob_vec.rs:221:9
  15: core::ptr::drop_in_place<bevy_ecs::storage::blob_vec::BlobVec>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  16: core::ptr::drop_in_place<bevy_ecs::storage::table::Column>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  17: core::ptr::drop_in_place<[bevy_ecs::storage::table::Column]>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  18: <alloc::vec::Vec<T,A> as core::ops::drop::Drop>::drop
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/vec/mod.rs:2810:13
  19: core::ptr::drop_in_place<alloc::vec::Vec<bevy_ecs::storage::table::Column>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  20: core::ptr::drop_in_place<bevy_ecs::storage::sparse_set::SparseSet<bevy_ecs::component::ComponentId,bevy_ecs::storage::table::Column>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  21: core::ptr::drop_in_place<bevy_ecs::archetype::Archetype>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  22: core::ptr::drop_in_place<[bevy_ecs::archetype::Archetype]>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  23: <alloc::vec::Vec<T,A> as core::ops::drop::Drop>::drop
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/vec/mod.rs:2810:13
  24: core::ptr::drop_in_place<alloc::vec::Vec<bevy_ecs::archetype::Archetype>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  25: core::ptr::drop_in_place<bevy_ecs::archetype::Archetypes>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  26: core::ptr::drop_in_place<bevy_ecs::world::World>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  27: core::ptr::drop_in_place<bevy_app::app::App>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  28: core::ptr::drop_in_place<bevy_winit::winit_runner_with::{{closure}}>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  29: core::ptr::drop_in_place<core::cell::UnsafeCell<dyn core::ops::function::FnMut<(winit::event::Event<()>,&winit::event_loop::EventLoopWindowTarget<()>,&mut winit::event_loop::ControlFlow)>+Output = ()>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  30: core::ptr::drop_in_place<core::cell::RefCell<dyn core::ops::function::FnMut<(winit::event::Event<()>,&winit::event_loop::EventLoopWindowTarget<()>,&mut winit::event_loop::ControlFlow)>+Output = ()>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  31: <alloc::rc::Rc<T> as core::ops::drop::Drop>::drop
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/rc.rs:1466:17
  32: core::ptr::drop_in_place<alloc::rc::Rc<core::cell::RefCell<dyn core::ops::function::FnMut<(winit::event::Event<()>,&winit::event_loop::EventLoopWindowTarget<()>,&mut winit::event_loop::ControlFlow)>+Output = ()>>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  33: core::ptr::drop_in_place<core::option::Option<alloc::rc::Rc<core::cell::RefCell<dyn core::ops::function::FnMut<(winit::event::Event<()>,&winit::event_loop::EventLoopWindowTarget<()>,&mut winit::event_loop::ControlFlow)>+Output = ()>>>>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ptr/mod.rs:188:1
  34: core::mem::drop
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/mem/mod.rs:909:24
  35: winit::platform_impl::platform::event_loop::EventLoop<T>::run_return::{{closure}}
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:191:17
  36: objc::rc::autorelease::autoreleasepool
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/objc-0.2.7/src/rc/autorelease.rs:29:5
  37: winit::platform_impl::platform::event_loop::EventLoop<T>::run_return
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:178:9
  38: winit::platform_impl::platform::event_loop::EventLoop<T>::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/platform_impl/macos/event_loop.rs:157:9
  39: winit::event_loop::EventLoop<T>::run
             at /Users/robparrett/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.26.1/src/event_loop.rs:154:9
  40: bevy_winit::run
             at ./crates/bevy_winit/src/lib.rs:180:5
  41: bevy_winit::winit_runner_with
             at ./crates/bevy_winit/src/lib.rs:604:9
  42: bevy_winit::winit_runner
             at ./crates/bevy_winit/src/lib.rs:220:5
  43: core::ops::function::Fn::call
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ops/function.rs:70:5
  44: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/alloc/src/boxed.rs:1868:9
  45: bevy_app::app::App::run
             at ./crates/bevy_app/src/app.rs:130:9
  46: headless_defaults::main
             at ./examples/app/headless_defaults.rs:4:5
  47: core::ops::function::FnOnce::call_once
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ops/function.rs:227:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

## Additional information

Discovered by [my automated example runner](https://github.com/rparrett/prototype_bevy_example_runner)

### PR DESCRIPTION
**Pull request by Shatur**
# Objective

Fixes #4440.

## Solution

Check if `RenderDevice` exists and add CI validation.

### CODE REVIEW

**Comment by superdump** [comment]
Is the headless_defaults file referred to somewhere else or was it missed from this PR?

**Comment by mockersf** [comment]
it trigger running the example in CI

**Review by mockersf — APPROVED** [review]

File: crates/bevy_pbr/src/light.rs [hunk 1]
```diff
@@ -712,8 +712,13 @@ pub(crate) fn assign_lights_to_clusters(
     lights_query: Query<(Entity, &GlobalTransform, &PointLight, &Visibility)>,
     mut lights: Local<Vec<PointLightAssignmentData>>,
     mut max_point_lights_warning_emitted: Local<bool>,
-    render_device: Res<RenderDevice>,
+    render_device: Option<Res<RenderDevice>>,
 ) {
+    let render_device = match render_device {
+        Some(render_device) => render_device,
+        None => return,
```
Comment (superdump) [inline crates/bevy_pbr/src/light.rs:719]: I wish there were a nicer syntax for doing this, like a ? operator that just returns if none else unwraps it. But I’m not aware of any.

**Review by superdump — APPROVED** [review]

File: crates/bevy_pbr/src/light.rs (hunk 1 above)
Comment (inquisitivecrystal) [reply crates/bevy_pbr/src/light.rs:719]: [Let else](rust-lang/rfcs#3137) will help with this slightly.

**Pull request closed without merging**